pyposeidon/mgmsh.py

```diff
@@ -635,34 +634,53 @@ def gmsh_execute(**kwargs):
         else:
             dim = -2
 
+        gmsh_args = kwargs.get("gmsh_args", {})
+
+        gmsh_args.update({"tol": 1e-20})
+
+        gargs = ""
+        for k, v in list(gmsh_args.items()):
+            if k[0] != "-":
+                k = "-" + k
+            gargs = gargs + " ".join([k, str(v)]) + " "
+
         # execute gmsh
-        with subprocess.Popen(
-            #            ["{} -bin -tol {} {} {}".format(bin_path, 1e-20, dim, "mymesh.geo")],
-            ["{} -tol {} {} {}".format(bin_path, 1e-20, dim, "mymesh.geo")],
+        ex = subprocess.run(
+            ["{} {} {} {}".format(bin_path, gargs, dim, "mymesh.geo")],
+            check=False,
+            capture_output=True,
+            text=True,
             cwd=calc_dir,
             shell=True,
-            stderr=subprocess.PIPE,
-            stdout=subprocess.PIPE,
             universal_newlines=True,
             # bufsize=1,
-        ) as ex:
-            output = ex.stdout.read()
-            error = ex.stderr.read()
+        )
+
+        with open(os.path.join(calc_dir, "myerr.log"), "w") as fd:
+            fd.write(ex.stderr)
+        with open(os.path.join(calc_dir, "myrun.log"), "w") as fd:
+            fd.write(ex.stdout)
 
-            lines = [l for l in error.splitlines() if l and l[:4] not in ["Info", "Warn"]]
+        lines = [l for l in ex.stderr.splitlines() if l and l[:4] not in ["Info", "Warn"]]
```

Comment:
I would move this line to line 675. The if/else immediately after this does not have to do anything with `lines`